jit.c

```diff
@@ -729,7 +729,7 @@ static void emit_op_system(rv_buffer *buff, uint32_t insn, struct riscv_t *rv)
     case 1: { /* CSRRW    (Atomic Read/Write CSR) */
         COMMENT("CSRRW");
         CODE("tmp_u32 = csr_csrrw(rv, %d, rv->X[%u]);\n", csr, rs1);
-        CODE("rv->X[%u] = rd ? tmp_u32 : rv->X[%u];\n", rd, rd);
+        CODE("rv->X[%u] = %u ? tmp_u32 : rv->X[%u];\n", rd, rd, rd);
```

Comment:
If rd equals to `rv_reg_zero` (i.e., `x0`), it is NOP. You should exclude such case. Send pull request(s) for RV32C and zero register handling later.